src/javaRosa/util.js

```diff
@@ -47,16 +47,22 @@ define([
         return getDefaultItextRoot(mug) + "-" + property;
     };
 
-    var looksLikeMarkdown = function(val) {
+    var looksLikeMarkdown = function(val, supportTables) {
         /* Regex checks (in order):
          * ordered lists
          * unordered lists
          * strikethrough
          * headings
          * italics/bold/bold italics
          * links
+         * tables (regex written after ||)
          */
-        return /^\d+[\.\)] |^\* |~~.+~~|# |\*{1,3}\S.*\*{1,3}|\[.+\]\(\S+\)/m.test(val);
+        var hasMarkdown = /^\d+[\.\)] |^\* |~~.+~~|# |\*{1,3}\S.*\*{1,3}|\[.+\]\(\S+\)/m.test(val),
+            hasTable = false;
+        if (supportTables) {
+            hasTable = /^(\|[^\n]+\|\r?\n)((?:\|:?[-]+:?)+\|)(\n(?:\|[^\n]+\|\r?\n?)*)?$/m.test(val);
```

Comment:
@AmitPhulera This doesn't pick up tables that aren't on the first line, like this one:
```
Look at my table
| col1      | col2 |
| ----------- | ----------- |
| data1      | data2       |
```
Could you update this?

Reply:
Thanks, @orangejenny for the feedback.

```
| col1      | col2 |
| ----------- | ----------- |
| data1      | data2       |
```

This syntax is not supported yet, I just tried to support the format that was there in the Mobile PR, but it is fairly easy to do. And I have not thought about the text in the beginning though 🤔 I will make the required changes.

Reply:
That syntax seems to work: https://www.commcarehq.org/a/bosco/apps/view/b7d394e5700f41d994335e585829891d/form/cb777bf09e38efb22975e46f4c9bc0d0511f4b37/source/#form/question1

Reply:
This one was a tricky one to figure out. So what's actually happening is that if you will paste 
```
| Column1      | Column2 |
| ----------- | ----------- |
| 100      | 200     |
| 110   | 215        |
```
to an input field which is already configured to be a markdown field then it will render a table because it's a valid table format and markdown-it render it as a table which is probably what happened in your case.

However, if you paste it directly to a new question or to an input field which is not a markdown field [then it will not render a table](https://www.commcarehq.org/a/bosco/apps/view/b7d394e5700f41d994335e585829891d/form/cb777bf09e38efb22975e46f4c9bc0d0511f4b37/source/#form/question_2) because the regex that I have used is not written to catch the spaces after `|` in the lines below the headers. 


And the formats that will work presently are 
```
| col1    | col2    | col3 |
|----|----|----|
| r1c1 | r1c2 | r1c3 |
| r2c1 | r2c2 | r2c3 |
```
```
| col1    | col2    | col3 |
|:----:|:----:|:----:|
| r1c1 | r1c2 | r1c3 |
| r2c1 | r2c2 | r2c3 |
```

I am updating the regex to support the third one too. And it actually [supports text with table,](https://www.commcarehq.org/a/bosco/apps/view/b7d394e5700f41d994335e585829891d/form/cb777bf09e38efb22975e46f4c9bc0d0511f4b37/source/#form/question_3) as the regex matches the table and text, is ignored.

@orangejenny  I have added two questions in the form, I am not sure if we should remove them afterward. Let me know if we have to remove them.

Reply:
Raised the PR https://github.com/dimagi/Vellum/pull/978

Reply:
Ah, I see. Thank you for the thorough explanation 🙂 

And no worries about changing the form, it's fine to play around with that one.